src/entities/fractions/currency_amount.rs

```diff
@@ -31,37 +37,43 @@ impl<T: CurrencyTrait> CurrencyAmount<T> {
     }
 
     // Returns a new currency amount instance from the unitless amount of token (raw amount)
-    pub fn from_raw_amount(currency: T, raw_amount: impl Into<BigInt>) -> CurrencyAmount<T> {
-        Self::new(currency, raw_amount, 1)
+    pub fn from_raw_amount(
+        currency: T,
+        raw_amount: impl Into<BigInt>,
+    ) -> Result<CurrencyAmount<T>, Error> {
+        Self::new(currency, raw_amount, 1).map_err(|err| Error::CreationError(format!("{}", err)))
```

Comment:
```suggestion
        Self::new(currency, raw_amount, 1)
```
It's not necessary to map error.

Reply:
idk, it's to transform the error into a custom type

Reply:
will remove majority of the maps

Reply:
or all of them, going through it again, they are truly not needed